### PR TITLE
feat: Support txpool.locals in txpool.cli

### LINKED ISSUE #5540: Add `--txpool.locals` Vec address to txpool cli args
### Describe the feature

Very similar to https://github.com/paradigmxyz/reth/issues/5520

but this time it should treat the given Addresses as local

 --txpool.locals value         
          Comma separated accounts to treat as locals

cc @Blaeaea

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by supernovahs**
closes #5540

### CODE REVIEW

File: bin/reth/src/args/txpool_args.rs [hunk 1]
```diff
@@ -70,7 +75,10 @@ impl RethTransactionPoolConfig for TxPoolArgs {
     /// Returns transaction pool configuration.
     fn pool_config(&self) -> PoolConfig {
         PoolConfig {
-            local_transactions_config: LocalTransactionConfig { no_exemptions: self.no_locals },
+            local_transactions_config: LocalTransactionConfig {
+                no_exemptions: self.no_locals,
+                local_addresses: self.locals.clone(),
+            },
```
Comment (supernovahs) [inline bin/reth/src/args/txpool_args.rs:81]: Let me know if this condition  is correct

File: crates/transaction-pool/src/pool/txpool.rs [hunk 2]
```diff
@@ -1253,7 +1253,13 @@ impl<T: PoolTransaction> AllTransactions<T> {
...
         transaction: ValidPoolTransaction<T>,
     ) -> Result<ValidPoolTransaction<T>, InsertErr<T>> {
-        if !transaction.origin.is_local() || self.local_transactions_config.no_local_exemptions() {
+        if !transaction.origin.is_local() ||
+            self.local_transactions_config.no_local_exemptions() ||
+            (self.local_transactions_config.has_local_address() &&
+                !self
+                    .local_transactions_config
+                    .contains_local_address(transaction.transaction.sender()))
+        {
```
Comment (supernovahs) [inline crates/transaction-pool/src/pool/txpool.rs:1262]: Let me know if this condition is correct

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/transaction-pool/src/validate/eth.rs [hunk 3]
```diff
@@ -237,7 +237,9 @@ where
 
         // Drop non-local transactions with a fee lower than the configured fee for acceptance into
         // the pool.
-        if (!origin.is_local() || self.local_transactions_config.no_local_exemptions()) &&
+        if (!origin.is_local() ||
+            self.local_transactions_config.no_local_exemptions() ||
+            self.local_transactions_config.contains_local_address(transaction.sender())) &&
```
Comment (supernovahs) [inline crates/transaction-pool/src/validate/eth.rs:242]: When it was  ` !self.local_transactions_config.contains_local_address(transaction.sender())) ` , received max slots error 
But this is also not correct even though tests pass

File: crates/transaction-pool/src/pool/txpool.rs [hunk 4]
```diff
@@ -1253,14 +1253,19 @@ impl<T: PoolTransaction> AllTransactions<T> {
         &self,
         transaction: ValidPoolTransaction<T>,
     ) -> Result<ValidPoolTransaction<T>, InsertErr<T>> {
-        if !transaction.origin.is_local() || self.local_transactions_config.no_local_exemptions() {
+        if !transaction.origin.is_local() ||
+            self.local_transactions_config.no_local_exemptions() ||
+            self.local_transactions_config
+                .contains_local_address(transaction.transaction.sender())
```
Comment (mattsse) [inline crates/transaction-pool/src/pool/txpool.rs:1259]: I want to move these checks to a function in LocalTransactionConfig that accepts an Origin and then returns true if it is considered local

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/transaction-pool/src/pool/txpool.rs (hunk 4 above)
Comment (supernovahs) [reply crates/transaction-pool/src/pool/txpool.rs]: Ok, but still figuring the failing test

**Review by mattsse — APPROVED** [review]
thanks!

the issue was in the is_local check

**Comment by supernovahs** [comment]
thanks

**Pull request merged**